brainfuck/executor.go

```diff
@@ -0,0 +1,40 @@
+package brainfuck
+
+//Execute is the main function of program that takes as an argument
+//string in Brainfuck and execute every instruction from this string
+func Execute(programCode string) {
+
+	//currentStack contains slice for executing commands in loop
+	var currentStack = []startingLoop{}
+
+	//In this loop, depending on type of symbol from string, we push instruction
+	//in currentStack
+	for i := 0; i < len(programCode); i++ {
+		var currentInstruction = instructions[programCode[i]]
+
+		switch t := currentInstruction.(type) {
+		case startingLoop:
+			//appending loop into currentStack
+			currentStack = append([]startingLoop{t}, currentStack...)
+		case endingLoop:
+			currentStack[1].loopStack = append(currentStack[1].loopStack, currentStack[0])
+			//popping elements of currentStack
+			currentStack = currentStack[1:(len(currentStack) + 1)]
+			//appending instructions to currentStack but into the loopStack
+		case plus, minus, shiftingRight, shiftingLeft, writeChar, readChar:
+			currentStack[0].loopStack = append(currentStack[0].loopStack, currentInstruction)
```

Comment:
currentStack[0].loopStack

what if there is + or - somewhere in the loop? it mens we will take always the first element which sounds incorrect

Reply:
?